### PR TITLE
Harshad_Assignment_20_Done_40mins

### PR DESCRIPTION
**Pull request by HarshadBJun20**
->Assignment 20 Done.
->Program 1 screenshot of output:
image

->Program 2 screenshot of output:
image

->Program 3 screenshot of output:
image

->Program 4 screenshot of output:
image

Please review..

### CODE REVIEW

File: src/harshadBJun20/Assignment20/ArmstrongNumberProgram.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+			num = num / 10;
+			sum += (seperateNum*seperateNum*seperateNum);
+		}
+		return sum;
+	}
+	public static void main(String[] args) {
+		ArmstrongNumberProgram armstrongNumberProgram = new ArmstrongNumberProgram();
+		int num = 153;
+		int sum = armstrongNumberProgram.checkArmstrongNumber(num);
+		if(num==sum) {
```
Comment (AasthaJJun20) [inline src/harshadBJun20/Assignment20/ArmstrongNumberProgram.java]: The logic is absolutely correct, but I would suggest to write the if-else statements in checkArmstrongNumber method itself and not in main method

File: src/harshadBJun20/Assignment20/FindDuplicateNumbersFromIntegerArray.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+				}
+			}
+		}
+	}
+	public static void main(String[] args) {
+		FindDuplicateNumbersFromIntegerArray findDuplicateNumbersFromIntegerArray = new FindDuplicateNumbersFromIntegerArray();
+		int[] arr= {0,3,1,2,3};
+		findDuplicateNumbersFromIntegerArray.findDuplicateNumber(arr);
+	}
+}
```
Comment (AasthaJJun20) [inline src/harshadBJun20/Assignment20/FindDuplicateNumbersFromIntegerArray.java:21]: Perfect, code looks good to me.

File: src/harshadBJun20/Assignment20/FindMissingNumberFromIntegerArray.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+/*Assignment 20 : 28th Jun 2020
+
+Program 1: Find the missing number in integer array of 1 to 10?
+Hint: You have given an integer array which contains numbers from 1 to 10 but one number is missing, you need to write a Java program to find that missing number in an array.*/
+
+public class FindMissingNumberFromIntegerArray {
+	
+	int findMissingNumber(int[] arr1) {
+		int sum=0,sumOfArrayElements=0;
+		sum = ((arr1.length+1)*((arr1.length+1)+1)/2);
```
Comment (AasthaJJun20) [inline src/harshadBJun20/Assignment20/FindMissingNumberFromIntegerArray.java:11]: Nice logic, didn't think of this.

File: src/harshadBJun20/Assignment20/FindMissingNumberFromIntegerArray.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+		}
+		return (sum-sumOfArrayElements);
+	}
+	public static void main(String[] args) {
+		FindMissingNumberFromIntegerArray findMissingNumberFromIntegerArray = new FindMissingNumberFromIntegerArray();
+		int[] arr= {1,8,3,10,6,7,5,9,4};
+		int missingNum = findMissingNumberFromIntegerArray.findMissingNumber(arr);
+		System.out.println("Missing number in the given array is: "+missingNum);
+	}
+}
```
Comment (AasthaJJun20) [inline src/harshadBJun20/Assignment20/FindMissingNumberFromIntegerArray.java:24]: You can directly print the returned variable in 
Print statement, initialization of missingNum is not required.
Just write System.out.println("Missing number in the given array is: "+findMissingNumberFromIntegerArray.findMissingNumber(arr));

File: src/harshadBJun20/Assignment20/PrintReverseIntegerNumber.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+Input:- 12345
+Output:- 54321*/
+public class PrintReverseIntegerNumber {
+	void printReverseNumbers(int num) {
+		int seperateNum=0;
+		System.out.println("Reverse order Number: ");
+		while(num>0) {
+			seperateNum = num % 10;
+			num = num / 10;
+			System.out.print(seperateNum);
```
Comment (AasthaJJun20) [inline src/harshadBJun20/Assignment20/PrintReverseIntegerNumber.java]: You are just printing the reverse order of the given number, Please implement integer reverse logic - which is just a small extension to your logic.

**Review by AasthaJJun20 — CHANGES_REQUESTED** [review]
Please make the suggested changes.

File: src/harshadBJun20/Assignment20/PrintReverseIntegerNumber.java (hunk 5 above)
Comment (HarshadBJun20) [reply src/harshadBJun20/Assignment20/PrintReverseIntegerNumber.java]: Hi Aastha,
Changes as advised is done. Please review. Thanks for these improvements.

File: src/harshadBJun20/Assignment20/FindMissingNumberFromIntegerArray.java (hunk 4 above)
Comment (HarshadBJun20) [reply src/harshadBJun20/Assignment20/FindMissingNumberFromIntegerArray.java:24]: Hi Aastha,
Changes as advised is done. Please review. Thanks for these improvements.

File: src/harshadBJun20/Assignment20/ArmstrongNumberProgram.java (hunk 1 above)
Comment (HarshadBJun20) [reply src/harshadBJun20/Assignment20/ArmstrongNumberProgram.java]: Hi Aastha,
Changes as advised is done. Please review.

File: src/harshadBJun20/Assignment20/FindDuplicateNumbersFromIntegerArray.java (hunk 2 above)
Comment (HarshadBJun20) [reply src/harshadBJun20/Assignment20/FindDuplicateNumbersFromIntegerArray.java:21]: Thanks!

File: src/harshadBJun20/Assignment20/FindMissingNumberFromIntegerArray.java (hunk 3 above)
Comment (HarshadBJun20) [reply src/harshadBJun20/Assignment20/FindMissingNumberFromIntegerArray.java:11]: Thanks!

**Review by AasthaJJun20 — APPROVED** [review]
Code looks good to me.

**Pull request merged**